File: sol/solution/0001-0100/0028. Find the Index of the First Occurrence in a String/Solution.ex
defmodule Solution do
  @spec str_str(haystack :: String.t(), needle :: String.t()) :: integer
  def str_str(haystack, needle) do
    cond do
      needle == "" ->
        0
      true ->
        case :binary.match(haystack, needle) do
          :nomatch -> -1
          {pos, _len} -> pos
        end
    end
  end
end

File: sol/solution/0001-0100/0028. Find the Index of the First Occurrence in a String/Solution.erl
-spec str_str(Haystack :: unicode:unicode_binary(),
              Needle   :: unicode:unicode_binary()) -> integer().
str_str(Haystack, Needle) ->
    case Needle of
        <<>> -> 0;
        _ ->
            case binary:match(Haystack, Needle) of
                nomatch -> -1;
                {Pos, _Len} -> Pos
            end
    end.